core/src/OpenMP/Kokkos_OpenMP_Instance.hpp

```diff
@@ -362,6 +365,48 @@ inline int OpenMP::impl_max_hardware_threads() noexcept {
   return Impl::g_openmp_hardware_max_threads;
 }
 
+namespace Experimental {
+namespace Impl {
+// Partitioning an Execution Space: expects space and integer arguments for
+// relative weight
+template <typename T>
+inline std::vector<OpenMP> create_OpenMP_instances(
+    OpenMP const& main_instance, std::vector<T> const& weights) {
+  static_assert(
+      std::is_arithmetic<T>::value,
+      "Kokkos Error: partitioning arguments must be integers or floats");
+  std::vector<OpenMP> instances(weights.size());
+  T total_weight = std::accumulate(weights.begin(), weights.end(), 0);
+  int const main_pool_size =
+      main_instance.impl_internal_space_instance()->thread_pool_size();
+
+  int resources_used = 0;
+  for (unsigned int i = 0; i < weights.size() - 1; ++i) {
+    int instance_pool_size = (weights[i] / total_weight) * main_pool_size;
```

Comment:
What if it is `0`?  Are we delegating this to the instance constructor?
If that is not legal (haven't yet looked at the constructor) should we raise a warning/an error here to produce better diagnostic for the user?

Reply:
We should do something for. I just realized that there is also a problem if you pass an empty weights vector

Reply:
The more I think about it, the more I am convinced that we should have a generic version that validates the input and then delegate to backend-specific implementations.  That way we can guarantee uniformity.  (Not asking you to do that as part of this PR.)

Reply:
The other backends do not care about the inputs. They are thrown away. The only thing that matters is the number of weights

Reply:
Do you also want to check for `total_weight` not to be zero? Maybe, just check `weights[i]` instead of `instance_pool_size`?